types/user.hh

```diff
@@ -51,6 +51,7 @@ public:
     bytes_view field_name(size_t i) const { return _field_names[i]; }
     sstring field_name_as_string(size_t i) const { return _string_field_names[i]; }
     const std::vector<bytes>& field_names() const { return _field_names; }
+    std::optional<size_t> idx_of_field(const bytes& name) const;
     sstring get_name_as_string() const;
```

Comment:
Why size_t? This isn't defined by the size of a pointer on the machine it is running on, but is a fixed limit.

Reply:
What else should I use?
> This isn't defined by the size of a pointer on the machine it is running on

that's not how I look at `size_t`. For me it's just a type to represent sizes and often indexes. The functions above: `field_name`, `field_name_as_string` also take `size_t` as an index type.

Reply:
size_t is used for data structures that can grow as much as main memory can. They're a bad choice when something doesn't depend on main memory. Since the problem already exists, we can keep it.

A good choice here is uint32_t.